drivers/timer/ite_it8xxx2_timer.c

```diff
@@ -304,10 +317,13 @@ int sys_clock_driver_init(const struct device *dev)
 
 	ARG_UNUSED(dev);
 
+	/* Enable 32-bit free run timer overflow interrupt */
+	IRQ_CONNECT(FREE_RUN_TIMER_IRQ, 0, free_run_timer_overflow_isr, NULL,
+			FREE_RUN_TIMER_FLAG);
 	/* Set 32-bit timer4 for free run*/
 	ret = timer_init(FREE_RUN_TIMER, EXT_PSR_32P768K, TRUE,
 			 FREE_RUN_TIMER_MAX_CNT, TRUE, FREE_RUN_TIMER_IRQ,
-			 FREE_RUN_TIMER_FLAG, FALSE, TRUE);
+			 FREE_RUN_TIMER_FLAG, TRUE, TRUE);
```

Comment:
As a follow on PR, can you update the timer_init() definition to use enums in place of the `raw`, `first_time_enable`, `with_int`, and `start` parameters?

This call might would look like:
```
timer_init(FREE_RUN_TIMER, EXT_PSR_32P768K, RAW_TYPE,
    FREE_RUN_TIMER_MAX_CNT, FIRST_TIME_ENABLE, FREE_RUN_TIMER_IRQ,
    FREE_RUN_TIMER_FLAG, WITH_TIMER_INT, START_TIMER)
```

Reply:
Ack.  @RuibinChang, Can you create a change to make these parameters clear?

Reply:
Ok